### PR TITLE
Fix wso2/product-is#3822: Log Generation for OAuth2 Introspection Endpoint

### LINKED ISSUE #3822: Log Generation for OAuth2 Introspection Endpoint
The logs should include the below information,
1. client id
2. grant type
3. scopes
4. user
5. issued at
6. expires_in_seconds
7. Token introspection latency
8. success
9. type
10. error
11. token - Optional, can enable via Event Listener property for OAuthTokenIssuanceLogPublisher. `<Property name="Token.Logging">true</Property>`

### PR DESCRIPTION
**Pull request by sarubi**
### Proposed changes in this pull request
Fix https://github.com/wso2/product-is/issues/3822
Should be merge after merging and building [https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/966](url)

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

**Review by IsuraD — APPROVED** [review]

File: components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/internal/OAuthDataPublisherServiceComponent.java [hunk 1]
```diff
@@ -118,4 +124,13 @@ protected void unsetOauthEventInterceptor(OAuthEventInterceptor oAuthEventInterc
         OAuthDataPublisherServiceHolder.getInstance().removeOauthEventListener(oAuthEventInterceptor);
     }
 
+    protected void setIdentityCoreInitializedEvent(IdentityCoreInitializedEvent identityCoreInitializedEvent) {
+
+        // Nothing to implement
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/internal/OAuthDataPublisherServiceComponent.java:129]: Missing full-stop at the end of the comment

File: components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/internal/OAuthDataPublisherServiceComponent.java [hunk 2]
```diff
@@ -118,4 +124,13 @@ protected void unsetOauthEventInterceptor(OAuthEventInterceptor oAuthEventInterc
...
+    }
+
+    protected void unsetIdentityCoreInitializedEvent(IdentityCoreInitializedEvent identityCoreInitializedEvent) {
+
+        // Nothing to implement
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/internal/OAuthDataPublisherServiceComponent.java:134]: Missing full-stop at the end of the comment

File: components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/listener/OAuthTokenIssuanceLogPublisher.java [hunk 3]
```diff
@@ -245,9 +265,119 @@ private void addStringToMap(String name, String value, Map<String, Object> param
         }
     }
 
-    private void logTransactionInfo(String info) {
+    /**
+     * printing the log for given log info type
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/listener/OAuthTokenIssuanceLogPublisher.java:269]: Start with the capital letter and end with the full-stop.

**Comment by sarubi** [comment]
@pamodaaw addressed with https://github.com/wso2-extensions/identity-data-publisher-oauth/pull/74

File: components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/listener/OAuthTokenIssuanceLogPublisher.java (hunk 3 above)
Comment (sarubi) [reply components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/listener/OAuthTokenIssuanceLogPublisher.java:269]: fixed with https://github.com/wso2-extensions/identity-data-publisher-oauth/pull/74

**Pull request merged**